extensions/oidc-rest-client/deployment/src/test/java/io/quarkus/oidc/rest/client/KeycloakRealmResourceManager.java

```diff
@@ -0,0 +1,141 @@
+package io.quarkus.oidc.rest.client;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import org.keycloak.representations.AccessTokenResponse;
+import org.keycloak.representations.idm.ClientRepresentation;
+import org.keycloak.representations.idm.CredentialRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.representations.idm.RoleRepresentation;
+import org.keycloak.representations.idm.RolesRepresentation;
+import org.keycloak.representations.idm.UserRepresentation;
+import org.keycloak.util.JsonSerialization;
+
+import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
+import io.restassured.RestAssured;
+
+public class KeycloakRealmResourceManager implements QuarkusTestResourceLifecycleManager {
```

Comment:
At some point we should look at consolidating these into some kind of common utility. It could actually be useful for end users as well to have an easy way to setup keycloak for tests.

Reply:
@stuartwdouglas +1, @cemnura is helping out with improving the test support for OIDC so we can work with him on it

Reply:
@stuartwdouglas see https://github.com/quarkusio/quarkus/issues/13957